### ISSUE TITLE
Unable to reproduce variational_autoencoder.py results

### ISSUE THREAD

**Issue by EmilienDupont** [fchollet/keras#4120]
Hi,

I have been experimenting with the VAE code example. However, I am unable to reproduce digits with the same quality as the ones described in the [blog](https://blog.keras.io/building-autoencoders-in-keras.html).

Specifically, running the exact code in variational_autoencoder.py I obtain
![screen shot 2016-10-19 at 13 59 39](https://cloud.githubusercontent.com/assets/12076034/19537882/87214772-9606-11e6-9968-75b6a9dbcf42.png)

If I reduce the range of points sampled from 15 stds to 2 stds the results are better
![screen shot 2016-10-19 at 14 17 26](https://cloud.githubusercontent.com/assets/12076034/19537944/bc4b690a-9606-11e6-86b0-a459804fa72f.png)

but still not nearly as good as the ones presented in the blog post
![screen shot 2016-10-19 at 14 00 03](https://cloud.githubusercontent.com/assets/12076034/19537961/caafe5ca-9606-11e6-83c6-7e48df3749ae.png)

For example, in the top right corner, the blob produced does not resemble a digit.

Do you have any idea why this is not working as well even though I am running the exact same code?

Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by EmilienDupont** [comment]
I am using tensorflow 0.10.0 and keras 1.1.0, running on a Tesla K80

**Comment by varoudis** [comment]
I could not reproduce some of the AE exmaples in the blog, i added an issue at the blog Github
Loss was quite higher and unable to reach that low values

**Comment by thangbui** [comment]
PR #4211 fixed this problem.
